src/core/colorspace.js

```diff
@@ -565,8 +565,15 @@ class IndexedCS extends ColorSpace {
     return this.base.getOutputLength(inputLength * this.base.numComps, alpha01);
   }
 
-  isDefaultDecode(decodeMap) {
-    return true; // Indexed color maps shouldn't be changed.
+  isDefaultDecode(decodeMap, bpc) {
+    if (!Array.isArray(decodeMap)) {
+      return true;
+    }
+    if (decodeMap.length !== 2) {
+      warn('Decode map length is not correct');
+      return true;
+    }
```

Comment:
Perhaps it would also be a good idea to check that the `bpc` parameter is an integer, larger than zero, in *addition* to the checks done above?